src/main/java/com/github/redouane59/twitter/helpers/RequestHelperV2.java

```diff
@@ -56,6 +63,48 @@
         return Optional.ofNullable(result);
     }
 
+    public void executeGetRequestWithConsumer(String url, Consumer consumer){
```

Comment:
Any feedback @hypr2771 ?

Reply:
## Blocking

Sure, I am wondering whether this is a good way, since you are most likely blocking the thread.
This looks OK since you directly call the `Consumer.accept`, but I am pretty sure _most of the time_ of this method is spent waiting.
I would have launch the request in an async way if your HTTP client supports it; or with an `Executor` which launches a `Runnable` (this is made easy using JDK 8 functional paradigm support).
This way, you are not blocking the code launching `executeGetRequestWithConsumer`; and it should be fine since the method returns a `void`, thanks to the `Consumer.accept` logic.

In other words, you should try to open this whole code in a new thread, and leave the thread call the Consumer.

## Generics

You are not telling the caller which type of `Consumer` he should provide. This is an anti-pattern for it may bring runtime exception which should be prevented during compilation.

## Naming

Once again, I feel that this method name does not have to be this noisy.
The fact that one argument is a `Consumer` should allow you to remove the `withConsumer` part.

Regarding the `execute`, your method is a _procedure_ (does things then exits without returning any value), hence it is implicitely "executing" something.

Finally, `get` is usually a keyword giving an advice that your method will return the following words.
Hence, I would prefer to call this method doing something like: `requestHelper.get(url, tweet -> accept(tweet))`.

Overall, this leaves us with a much succinct method name (which is always preferable):

```suggestion
    public void get(String url, Consumer<TweetDTOv2> consumer){
```

Reply:
Fixed for `Generics` & `Naming` parts thanks, I'm going to check how to make it async tonight.

Reply:
Done for the `async` part using `call.enqueue(new Callback() { ...`